File: helium/src/main/groovy/com/stanfy/helium/handler/codegen/objectivec/ObjCHeaderFile.java
package com.stanfy.helium.handler.codegen.objectivec;

/**
 * Created by ptaykalo on 8/17/14.
 * Class that represents objective-C header file
 * Will always have .h extension
 */
public class ObjCHeaderFile extends ObjCFile {

  public ObjCHeaderFile(final String name) {
    super(name);
  }

  @Override
  public String getExtension() {
    return "h";
  }
}

File: codegen/objc/objc-entities/src/main/kotlin/com/stanfy/helium/handler/codegen/objectivec/entity/filetree/ObjCHeaderFile.kt
package com.stanfy.helium.handler.codegen.objectivec.entity.filetree;

import com.stanfy.helium.handler.codegen.objectivec.entity.filetree.ObjCFile

/**
 * Created by ptaykalo on 8/17/14.
 * Class that represents objective-C header filetree
 * Will always have .h extension
 */
public class ObjCHeaderFile(name: String, contents: String) : ObjCFile(name, contents) {

  public constructor(name: String) : this(name, "") {
  }

  override fun getExtension(): String {
    return "h";
  }
}